pkg/controller/instance/instance_controller.go

```diff
@@ -305,57 +305,42 @@ func instanceEventFilter(mgr manager.Manager) predicate.Funcs {
 	}
 }
 
-func createPlan(r *ReconcileInstance, planName string, instance *kudov1alpha1.Instance) error {
+func createPlan(c client.Client, r record.EventRecorder, scheme *runtime.Scheme, planName string, instance *kudov1alpha1.Instance) error {
 	ctx := context.TODO()
 
-	planExecution := newPlanExecution(instance, planName, r.scheme)
-	return createPlanExecution(ctx, instance, planExecution, r, planName)
-}
-
-// createPlanExecution takes all the k8s objects needed to create the actual plan
-func createPlanExecution(ctx context.Context, instance *kudov1alpha1.Instance, plan kudov1alpha1.PlanExecution, r *ReconcileInstance, planName string) error {
-	log.Printf("Creating PlanExecution of plan %s for instance %s", planName, instance.Name)
-	r.recorder.Event(instance, "Normal", "CreatePlanExecution", fmt.Sprintf("Creating \"%v\" plan execution", planName))
+	planExecution := newPlanExecution(instance, planName, scheme)
+	log.Printf("Creating PlanExecution of planExecution %s for instance %s", planName, instance.Name)
+	r.Event(instance, "Normal", "CreatePlanExecution", fmt.Sprintf("Creating \"%v\" planExecution execution", planName))
 
 	// Make this instance the owner of the PlanExecution
-	if err := controllerutil.SetControllerReference(instance, &plan, r.scheme); err != nil {
+	if err := controllerutil.SetControllerReference(instance, planExecution, scheme); err != nil {
 		log.Printf("InstanceController: Error setting ControllerReference")
 		return err
 	}
-	if err := r.Create(ctx, &plan); err != nil {
-		log.Printf("InstanceController: Error creating planexecution \"%v\": %v", plan.Name, err)
-		r.recorder.Event(instance, "Warning", "CreatePlanExecution", fmt.Sprintf("Error creating planexecution \"%v\": %v", plan.Name, err))
+	if err := c.Create(ctx, planExecution); err != nil {
+		log.Printf("InstanceController: Error creating planexecution \"%v\": %v", planExecution.Name, err)
+		r.Event(instance, "Warning", "CreatePlanExecution", fmt.Sprintf("Error creating planexecution \"%v\": %v", planExecution.Name, err))
 		return err
 	}
-	log.Printf("Created PlanExecution of plan %s for instance %s", planName, instance.Name)
-	r.recorder.Event(instance, "Normal", "PlanCreated", fmt.Sprintf("PlanExecution \"%v\" created", plan.Name))
-	return nil
-}
+	log.Printf("Created PlanExecution of planExecution %s for instance %s", planName, instance.Name)
+	r.Event(instance, "Normal", "PlanCreated", fmt.Sprintf("PlanExecution \"%v\" created", planExecution.Name))
 
-//todo: REMOVE this... we do not want to create a plan outside of reconcile
-// most likely... any use of this function is wrong... all plans should be created in reconcile
-func createPlanOld(mgr manager.Manager, planName string, instance *kudov1alpha1.Instance) error {
```

Comment:
it was real pain maintaining two, while the only difference was how they accepted the dependencies which was pretty easy to unite